assets/lokomotive-kubernetes/aws/flatcar-linux/kubernetes/controllers.tf

```diff
@@ -23,8 +87,9 @@ resource "aws_instance" "controllers" {
 
   instance_type = var.controller_type
 
-  ami       = local.ami_id
-  user_data = data.ct_config.controller-ignitions[count.index].rendered
+  ami                  = local.ami_id
+  user_data            = data.ct_config.controller-ignitions[count.index].rendered
+  iam_instance_profile = join("", aws_iam_instance_profile.csi-driver-instance-profile.*.name)
```

Comment:
I would use `aws_iam_instance_profile.csi-driver-instance-profile.[0].name`. There can't be more than one instance profile so `join` doesn't make sense to me here.

Reply:
This suggestion doesn't work when `set_csi_driver_iam_role = false`